src/Features/CSharp/Portable/Completion/KeywordRecommenders/StackAllocKeywordRecommender.cs

```diff
@@ -16,20 +16,41 @@ public StackAllocKeywordRecommender()
 
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
-            // type t = |
-            var token = context.TargetToken;
-            if (token.IsUnsafeContext())
+            var node = context.TargetToken.Parent;
+
+            if (node == null)
+            {
+                return false;
+            }
+
+            if (node.IsKind(SyntaxKind.ParenthesizedExpression) || node.IsKind(SyntaxKind.CastExpression))
+            {
+                node = node.Parent;
+            }
+
+            while (node.IsKind(SyntaxKind.ConditionalExpression))
+            {
+                node = node.Parent;
+            }
+
+            if (node.IsKind(SyntaxKind.SimpleAssignmentExpression))
+            {
+                return node.Parent.IsKind(SyntaxKind.ExpressionStatement);
```

Comment:
It might be nested inside other nodes, like casts or conditionals.

---
In reply to: [155678306](https://github.com/dotnet/roslyn/pull/23655#discussion_r155678306) [](ancestors = 155678306)